### ISSUE TITLE
Feature Request: property to customize file extension icon

### ISSUE THREAD

**Issue by lepinekong** [Microsoft/vscode#44026]
- VSCode Version: all
- OS Version: all

Can't there be a quick way to customize file extension icon instead of having to create a whole extension for a custom theme as explained here https://github.com/Microsoft/vscode/issues/10804?

It's really daunting to have to learn the whole system of extension just for this tiny need. 

Does this issue occur when all extensions are disabled?: Yes/No

**Comment by aeschli** [comment]
Can you tell more your use case? Do you don't like some of the icons, or you have some missing icons? That would be https://github.com/Microsoft/vscode/issues/14662

**Comment by ksmithut** [comment]
I'll put my use case here. I really like the seti-ui file icon theme, but I want folders to have just the plain folder icon. Right now they have no icon. I most definitely don't want to get into the business of maintaining an extension just for file-icon theme because I don't want to spend the time and resources required to stay up to date with the upstream theme and vscode api changes. I am willing to see my personal settings break because the workaround is remove my custom settings. I am not willing to have to go through the process of publishing a new version then downloading the new version of the plugin just to add folder icons to a theme that does not have folder icons.

**Comment by RobPurcellUK** [comment]
This would help for an internal system we have which uses YAML files with specified names.
We would like to be able to set the icon to use for the `.thing.yml` files we use (like the fileNames part of an iconTheme vscode extension, not just by file type).

**Comment by eponymz** [comment]
Being able to upload a certain icon (of your design or something found) and assign it to a certain file extension would be nice as well. Use case here is, for some projects I am on I use the confluence extension on the market. Would be nice to set my own custom icon for files with those extensions so they aren't just the default 'txt' icon.

**Comment by Hubro** [comment]
I just want my `Procfile.dev` file to have the same icon as `Procfile`. Is this currently possible?

**Comment by ksmithut** [comment]
@Hubro I believe you can do this in your user settings with the `files.associations`, something like this:

**Comment by Hubro** [comment]
@ksmithut I have actually associated `"Procfile*"` with YAML, which affects the syntax highlighting of both `Procfile` and `Procfile.dev`, but only the latter gets the YAML icon:

![image](https://user-images.githubusercontent.com/597206/61292716-3d01a700-a7d2-11e9-8359-5d47036034ed.png)

Procfile still has the Heroku icon.

**Comment by orthodoX** [comment]
I completely agree with @Hubro and @eponymz for this feature. For example, there is a `.leex` extension introduced few months ago for Phoenix web framework. I'd like to set for it the same icon that my theme uses for `.eex` extension, until it gets officially supported by theme(s). Imo it would be so nice to have this configurable in the `settings.json`.

**Comment by diminutivesloop** [comment]
I use the Tailwind CSS IntelliSense extension and when I enable the "Tailwind CSS" mode for css files some themes don't show the css icon for those files. This feature would allow me to force it back to the regular css icon.

**Comment by lewnelson** [comment]
Our use case is being able to map a file extension we are using for TypeScript test files to the orange TS icon within Seti. We're using `.e2e-spec.ts` as the file extension. It would be nice to be able to extend the existing theme via settings. For now though I think we will just rename the files.

**Comment by learn2reid** [comment]
Is there a solution for this yet?

**Comment by DecimalTurn** [comment]
@learn2reid, as far as I know, the only option to customize a single icon without having to make an new extension is to use an icon theme that supports customization. Unfortunatly, the default icon themes for VS Code don't support it, but you can install one that does such as: 

* [vscode-material-icon-theme](https://github.com/material-extensions/vscode-material-icon-theme#custom-icon-associations)
* [catppuccin/vscode-icons](https://github.com/catppuccin/vscode-icons#custom-icon-associations)
